manifests/plugin.pp

```diff
@@ -0,0 +1,43 @@
+# common code for the 3rd party plugins
+#
+define collectd::plugin (
+  $config_file_name,
+  $config_template,
+  $manage_package = true,
+  $package_name = undef,
+  $package_version = present,
+  $modules = undef,
+  $filter_metrics = false,
+  $filter_metric_rules = {},
+){
+  validate_string($config_file_name)
+  validate_string($config_template)
+  validate_bool($filter_metrics)
+
+  if $modules {
+    validate_hash($modules)
+  }
+
+  if $manage_package {
+    unless $package_name  {
+      $package_name = $name
+    }
+
+    collectd::check_and_install_package { "${package_name} for ${name} plugin":
```

Comment:
What about the other way around, split the deb packages up so you only install what you need, similar to how the RPM's are doing it currently. modifying the patch to accomidate for the current split behavior